supabase_auth/types.py

```diff
@@ -644,11 +652,17 @@ class AuthMFAEnrollResponse(BaseModel):
     """
     Friendly name of the factor, useful for distinguishing between factors
     """
-    phone: str
+    phone: Optional[str] = None
     """
     Phone number of the MFA factor in E.164 format. Used to send messages
     """
 
+    @model_validator_v1_v2_compat
+    def validate_phone_required_for_phone_type(cls, values: dict) -> dict:
+        if values.get("type") == "phone" and not values.get("phone"):
+            raise ValueError("phone is required when type is 'phone'")
+        return values
+
```

Comment:
@silentworks is this the correct way of doing it?

Reply:
Yeah that's correct.